packages/@aws-cdk/app-delivery/lib/pipeline-deploy-stack-action.ts

```diff
@@ -41,6 +41,50 @@ export interface PipelineDeployStackActionProps {
    * @default ``createChangeSetRunOrder + 1``
    */
   executeChangeSetRunOrder?: number;
+
+  /**
+   * IAM role to assume when deploying changes.
+   *
+   * If not specified, a fresh role is created. The role is created with zero
+   * permissions unless `fullPermissions` is true, in which case the role will have
+   * full permissions.
+   *
+   * @default A fresh role with full or no permissions (depending on the value of `fullPermissions`).
+   */
+  role?: iam.Role;
+
+  // tslint:disable:max-line-length Because of long URLs in documentation
+  /**
+   * Acknowledge certain changes made as part of deployment
+   *
+   * For stacks that contain certain resources, explicit acknowledgement that AWS CloudFormation
+   * might create or update those resources. For example, you must specify CAPABILITY_IAM if your
+   * stack template contains AWS Identity and Access Management (IAM) resources. For more
+   * information, see [Acknowledging IAM Resources in AWS CloudFormation Templates](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/using-iam-template.html#using-iam-capabilities).
+   *
+   * @default No capabitilities passed, unless `fullPermissions` is true
+   */
+  capabilities?: cfn.CloudFormationCapabilities;
+  // tslint:enable:max-line-length
+
+  /**
+   * Whether to grant full permissions to CloudFormation while deploying this template.
+   *
+   * Setting this to `true` affects the defaults for `role` and `capabilities`, if you
+   * don't specify any alternatives.
+   *
+   * The default role that will be created for you will have full (i.e., `*`)
+   * permissions on all resources, and the deployment will have named IAM
+   * capabilities (i.e., able to create all IAM resources).
+   *
+   * This is a shorthand that you can use if you fully trust the templates that
+   * are deployed in this pipeline. If you want more fine-grained permissions,
+   * use `addToRolePolicy` and `capabilities` to control what the CloudFormation
+   * deployment is allowed to do.
+   *
+   * @default false
+   */
+  fullPermissions?: boolean;
```

Comment:
I went with `adminPermissions` because when I put in `administrator` I asked my self of what? I thought `adminPermissions` would be more clear, if you are set on `administrator` let me know.